frontend/javascripts/dashboard/dataset/dataset_cache_provider.js

```diff
@@ -44,12 +56,16 @@ export const DatasetCacheContext = createContext<Context>({
   isLoading: false,
   fetchDatasets: async () => {},
   checkDatasets: async () => {},
-  updateDataset: async () => {},
+  reloadDataset: async () => {},
+  updateCachedDataset: async () => {},
 });
 
 export default function DatasetCacheProvider({ children }: { children: Node }) {
   const [datasets, setDatasets] = useState(datasetCache.get());
   const [isLoading, setIsLoading] = useState(false);
+  const [pendingDatasetUpdates, setPendingDatasetUpdates] = useState<{
+    [string]: Array<APIDataset>,
```

Comment:
This should probably be

```suggestion
    [string]: Promise<APIDataset>,
```

to fix the flow error.